### PR TITLE
Add missing QEOMResult docs

### LINKED ISSUE #1019: QEOMResult needs docs and other minor issues
In regards of the newly updated [QEOMResult](https://github.com/Qiskit/qiskit-nature/blob/00d85e3fcca3c405dbe8fd8027784ad2fbca3971/qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py#L916) 

1. What was all getter/setters before in the result many have been changed to public instance vars but there are no docs for these - consider using `Attributes` to document these. With getter/setters the type could be different - maybe the None aspect is really invalid so when this is documented, although the type now says | None, maybe this should not be exposed. I don't think setting any of these to None is really valid right, we would expect them all to be set. 

2. A public instance field `eigenvalues` has been added yet the parent EigensolverResult has this already and I think that should be used given its the same name.

3. In terms of some of the old setters (W, Q...) being deprecated. If code is still using these - presumably some other QEOM implementation building a result, it will not set S or H so will the result be valid - well S and H will end up being None so it will look like the older result. If these new fields are not being read then it will not be an issue I guess - it would only be the case where its being setup the old way but read by new code that now expects S and H to be set. I guess there is no easy way around this. Maybe we should add some documentation that explains things - though most users I think are going to just be reading the result not  creating one. In terms of the deprecation messages maybe the docs will help as they are at present for the setters they say something like `The W matrix is now computed from the S matrix` which may imply you should be setting S now but the messages could be more prescriptive e.g. `You should now set the S matrix from which W will be computed`. But maybe things will be clearer overall if this transition over to H and S is mentioned in some docs.

### PR DESCRIPTION
**Pull request by Anthony-Gandon**
### Summary

Fix #1019

- This PR adds the missing docstring of the QEOMResultClass.
- It also deprecates the M, Q, W, V std setters which had been forgotten.

### Details and comments

- [x] Remaining is the issue with the typing None for public attributes in the `QEOMResult`.

### CODE REVIEW

File: qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py
Comment (mrossinek) [inline qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py:972]: Referring to m, q, v, and w here does not work since these are not explained anywhere else. I would suggest we rather just refer to H and S, respectively, and properly define these somewhere here (probably in the few lines above explaining `h_matrix` and `s_matrix`.

File: releasenotes/notes/update-qeomresult-docstring-and-pending-deprecate-mqwv-matrix-std-setters-5bc2d84a941dca72.yaml
Comment (mrossinek) [inline releasenotes/notes/update-qeomresult-docstring-and-pending-deprecate-mqwv-matrix-std-setters-5bc2d84a941dca72.yaml:6]:
```suggestion
```
Since the previous changes were not released yet, this note is not necessary

**Review by mrossinek — COMMENTED** [review]
Some initial comments :+1:

File: qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py
Comment (Anthony-Gandon) [reply qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py:966]: I thought that the EigenstateResult.from_result() was requiring an attribute eigenstate from EigensolveResult. I was not aware that this behaviour was changed. (QEOM cannot produce the excited circuits but gives a proxy on how to evaluate observables on them)

File: releasenotes/notes/update-qeomresult-docstring-and-pending-deprecate-mqwv-matrix-std-setters-5bc2d84a941dca72.yaml
Comment (Anthony-Gandon) [reply releasenotes/notes/update-qeomresult-docstring-and-pending-deprecate-mqwv-matrix-std-setters-5bc2d84a941dca72.yaml:6]: Done

File: qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py
Comment (Anthony-Gandon) [reply qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py]: Done

**Review by mrossinek — DISMISSED** [review]
LGTM, thanks!

Since @woodsp-ibm raised the original issue, I will await a review from him before merging :+1:

File: qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py
Comment (woodsp-ibm) [inline qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py:956]: Can we add types to these fields - it goes in brackets after the name before the : (examples can be found elsewhere in nature). I think I would avoid putting None here - I don't think its really a valid value as such (unless it is in which case we should have it there) more that this is just as aspect of how the class is initialized one field at a time.

File: qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py
Comment (woodsp-ibm) [inline qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py:276]: Now QEOMResult, it seems, does not appear in the docs. I think the only place where we expose this to an end user is via the raw_result field of the EigenstateResult that is returned here from this solve method. So I thinking that we can expand this returns text to state that fact and link the QEOMResult class here so that it can be navigated to from here directly in the context where its relevant, rather than try to put it elsewhere like other classes where we have them documented and exposed via the init.

**Comment by Anthony-Gandon** [comment]
It seems that I got the linking to the QEOMResult wrong in the qeom.solve() because I cannot seem to make it clickable in the docs. Any clue on how to fix it?

**Comment by mrossinek** [comment]
> It seems that I got the linking to the QEOMResult wrong in the qeom.solve() because I cannot seem to make it clickable in the docs. Any clue on how to fix it?

I think it is probably just that type in the link which I suggested to correct above just now

File: qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py
Comment (mrossinek) [inline qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py:1039]: We could add a sentence here stating that the following attributes exist on this class. But I think it would also be fine as is.

**Review by mrossinek — DISMISSED** [review]
A minor comment but I would be fine with merging this as is, too. Thanks, Anthony!

File: releasenotes/notes/update-qeomresult-docstring-and-pending-deprecate-mqwv-matrix-std-setters-5bc2d84a941dca72.yaml
Comment (Anthony-Gandon) [reply releasenotes/notes/update-qeomresult-docstring-and-pending-deprecate-mqwv-matrix-std-setters-5bc2d84a941dca72.yaml]: Sure, done

File: releasenotes/notes/pending-deprecate-mqwv-matrix-setter-1cd9e2dfb68d1a85.yaml
Comment (woodsp-ibm) [inline releasenotes/notes/pending-deprecate-mqwv-matrix-setter-1cd9e2dfb68d1a85.yaml:5]:
```suggestion
    :class:`~qiskit_nature.second_q.algorithms.excited_state_solvers.qeom.QEOMResult`
```

Comment (woodsp-ibm) [reply releasenotes/notes/pending-deprecate-mqwv-matrix-setter-1cd9e2dfb68d1a85.yaml:5]: I think that typo prevents it coming out as a link

Comment (Anthony-Gandon) [reply releasenotes/notes/pending-deprecate-mqwv-matrix-setter-1cd9e2dfb68d1a85.yaml]: oops sry about that

**Review by woodsp-ibm — APPROVED** [review]
Thanks for fixing things @Anthony-Gandon - LGTM

**Pull request merged**
